javalin/src/main/java/io/javalin/http/JavalinServletHandler.kt

```diff
@@ -89,7 +96,7 @@ class JavalinServletHandler(
     /** Executes the given task with proper error handling and returns next task to execute as future */
     private fun executeNextTask(): CompletableFuture<InputStream> {
         val task = tasks.poll()
-        if (errored && task.stage.haltsOnError) {
+        if (errored && task.stage.haltsOnError || redirected) {
```

Comment:
I'd replace it with sth like 

```kotlin
if (errored && task.stage.haltsOnError || ctx.isTerminated()) { }

private fun Context.isTerminated(): Boolean =
   ctx.attribute<Boolean>("javalin-servlet-terminated") == true
```

In the future we could even consider exposing sth like `ctx.terminate()` in the official api to allow user immediately shutdown the execution.